src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java

```diff
@@ -60,6 +64,54 @@ public String getObjectPath() {
         return objectPath;
     }
 
+    @Override
+    public void addDevice(String uri) {
+        try {
+            m.addDeviceLink(new URI(uri));
+        } catch (IOException | InvalidKeyException e) {
+            throw new Error.Failure(e.getClass().getSimpleName() + " Add device link failed. " + e.getMessage());
+        } catch (URISyntaxException e) {
+            throw new Error.Failure(e.getClass().getSimpleName() + " Device link uri has invalid format: " + e.getMessage());
```

Comment:
Maybe add a new error class for this? `Error.InvalidUri`

Reply:
and activated :)